joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala

```diff
@@ -515,6 +513,44 @@ class SimpleAstCreationPassTest extends AbstractPassTest {
       accessElement.canonicalName shouldBe "foo"
     }
 
+    "be correct for call on object with argument" in AstFixture("""
+        |function method(x) {
+        |  a.b(x);
+        |}
+        """.stripMargin) { cpg =>
+      val List(method) = cpg.method.nameExact("method").l
+      val List(block)  = method.astChildren.isBlock.l
+
+      val List(call) = block.astChildren.isCall.l
+      call.code shouldBe "a.b(x)"
+      call.name shouldBe ""
+      call.dispatchType shouldBe DispatchTypes.DYNAMIC_DISPATCH
+
+      val List(receiver) = call.receiver.isCall.l
+      receiver.code shouldBe "a.b"
+      receiver.methodFullName shouldBe Operators.fieldAccess
+
+      val List(receiverAsArg) = call.argument(0).cast[Call].l
+      receiver shouldBe receiverAsArg
+
+      val List(base) = receiver.astChildren.isIdentifier.argumentIndex(1).l
+      base.name shouldBe "a"
+
+      val List(accessElement) = receiver.astChildren.isFieldIdentifier.argumentIndex(2).l
+      accessElement.canonicalName shouldBe "b"
+
+      val List(recvArg, baseArg, arg) = call.argument.l.sortBy(_.order)
+      recvArg shouldBe receiver
+      recvArg.argumentIndex shouldBe 0
+      recvArg.order shouldBe 0
+      baseArg.code shouldBe "a"
+      baseArg.argumentIndex shouldBe 0
+      baseArg.order shouldBe 1
```

Comment:
When tracking flows into parameter 0 of the callee of such a call, I think we'd get duplicate dataflows here - one through `recvArg` and another through `baseArg`.

My vote would be to just get rid of the `baseArg` - but I don't know the reason why we generated the `baseArg` in js2cpg to begin with.